src/spdx/parser/tagvalue/parser/tagvalue.py

```diff
@@ -0,0 +1,689 @@
+
+    @grammar_rule("file_conc : FILE_LICS_CONC license_or_no_assertion_or_none")
+    def p_file_license_concluded(self, p):
+        self.check_that_current_element_matches_class_for_value(File, p.lineno(1))
+        set_value(p, self.current_element, argument_name="license_concluded")
+
+    # parsing methods for package
+    @grammar_rule("pkg_attribution_text : PKG_ATTRIBUTION_TEXT error\n description : PKG_DESC error\n "
+                  "pkg_comment : PKG_COMMENT error\n summary : PKG_SUM error\n pkg_cr_text : PKG_CPY_TEXT error\n "
+                  "pkg_ext_ref : PKG_EXT_REF error\n pkg_lic_comment : PKG_LICS_COMMENT error\n "
+                  "pkg_lic_decl : PKG_LICS_DECL error\n pkg_lic_ff : PKG_LICS_FFILE error \n "
+                  "pkg_lic_conc : PKG_LICS_CONC error\n source_info : PKG_SRC_INFO error\n homepage : PKG_HOME error\n "
+                  "pkg_checksum : PKG_CHECKSUM error\n verification_code : PKG_VERF_CODE error\n "
+                  "download_location : PKG_DOWN error\n files_analyzed : PKG_FILES_ANALYZED error\n "
+                  "originator : PKG_ORIG error\n supplier : PKG_SUPPL error\n pkg_file_name : PKG_FILE_NAME error\n "
+                  "package_version : PKG_VERSION error\n primary_package_purpose : PRIMARY_PACKAGE_PURPOSE error\n "
+                  "built_date : BUILT_DATE error\n release_date : RELEASE_DATE error\n "
+                  "valid_until_date : VALID_UNTIL_DATE error")
+    def p_package_value_error(self, p):
+        self.current_element["logger"].append(
+            f"Error while parsing {p[1]}: Token did not match specified grammar rule. Line: {p.lineno(1)}")
+
+    @grammar_rule("description : PKG_DESC text_or_line\n summary : PKG_SUM text_or_line\n "
+                  "source_info : PKG_SRC_INFO text_or_line\n homepage : PKG_HOME line_or_no_assertion_or_none\n "
+                  "download_location : PKG_DOWN line_or_no_assertion_or_none\n "
+                  "originator : PKG_ORIG actor_or_no_assertion\n supplier : PKG_SUPPL actor_or_no_assertion")
+    def p_generic_package_value(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        set_value(p, self.current_element)
+
+    @grammar_rule("package_name : PKG_NAME LINE")
+    def p_package_name(self, p):
+        self.initialize_new_current_element(Package)
+        set_value(p, self.current_element, argument_name="name")
+
+    @grammar_rule("package_name : PKG_NAME error")
+    def p_package_name_error(self, p):
+        self.initialize_new_current_element(Package)
+        self.current_element["logger"].append(
+            f"Error while parsing {p[1]}: Token did not match specified grammar rule. Line: {p.lineno(1)}")
+
+    @grammar_rule("pkg_comment : PKG_COMMENT text_or_line")
+    def p_pkg_comment(self, p):
+        self.check_that_current_element_matches_class_for_value(Package, p.lineno(1))
+        set_value(p, self.current_element, argument_name="comment")
```

Comment:
Most of these rules are only separate from each other because the `argument_name` has to be set differently.
So here is a major refactoring suggestion: Put all those into `p_generic_package_value` and do the determination of `argument_name` there. Then, these last two lines above need only appear once.

Reply:
done